packages/salesforcedx-vscode-soql/test/jest/editor/soqlEditorProvider.test.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import {
+  BUILDER_VIEW_TYPE,
+  SOQL_BUILDER_WEB_ASSETS_PATH
+} from '../../../src/constants';
+import { HtmlUtils } from '../../../src/editor/htmlUtils';
+import { SOQLEditorInstance } from '../../../src/editor/soqlEditorInstance';
+import { SOQLEditorProvider } from '../../../src/editor/soqlEditorProvider';
+import { nls } from '../../../src/messages';
+import { channelService, isDefaultOrgSet } from '../../../src/sf';
+
+jest.mock('../../../src/sf', () => ({
+  ...jest.requireActual('../../../src/sf'),
+  isDefaultOrgSet: jest.fn()
+}));
+
+describe('SOQLEditorProvider', () => {
+  let extensionContext: vscode.ExtensionContext;
+  let registerCustomEditorProviderMock: jest.SpyInstance;
+  const mockDisposable = new vscode.Disposable(() => {});
+
+  beforeEach(() => {
+    extensionContext = {
+      extensionPath: 'path/to/extension',
+      subscriptions: [],
+      // eslint-disable-next-line @typescript-eslint/no-shadow
+      asAbsolutePath: jest.fn((path: string) => `/mocked/path/${path}`),
+      extension: {
+        packageJSON: {
+          soqlBuilderWebAssetsPath: ['path', 'to', 'soqlBuilder']
+        }
+      }
+    } as unknown as vscode.ExtensionContext;
+    registerCustomEditorProviderMock = (vscode.window
+      .registerCustomEditorProvider as jest.Mock) = jest
+      .fn()
+      .mockReturnValue(mockDisposable);
+    (isDefaultOrgSet as jest.Mock).mockReturnValue(true);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
```

Comment:
This shouldn't be necessary. I think have jest configured to clear after each test.